src/main/java/uk/gov/hmcts/ccd/domain/service/CreateOrUpdateUserProfileOperation.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.hmcts.ccd.domain.service;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ccd.data.jurisdiction.JurisdictionEntity;
+import uk.gov.hmcts.ccd.data.jurisdiction.JurisdictionRepository;
+import uk.gov.hmcts.ccd.data.userprofile.UserProfileRepository;
+import uk.gov.hmcts.ccd.domain.model.Jurisdiction;
+import uk.gov.hmcts.ccd.domain.model.UserProfile;
+import uk.gov.hmcts.ccd.endpoint.exception.BadRequestException;
+
+@Service
+public class CreateOrUpdateUserProfileOperation {
+
+    private static final Logger LOG = LoggerFactory.getLogger(CreateOrUpdateUserProfileOperation.class);
+    private final UserProfileRepository userProfileRepository;
+    private final JurisdictionRepository jurisdictionRepository;
+    private final CreateUserProfileOperation createUserProfileOperation;
+
+    @Autowired
+    public CreateOrUpdateUserProfileOperation(final UserProfileRepository userProfileRepository,
+                                              final JurisdictionRepository jurisdictionRepository,
+                                              final CreateUserProfileOperation createUserProfileOperation) {
+        this.userProfileRepository = userProfileRepository;
+        this.jurisdictionRepository = jurisdictionRepository;
+        this.createUserProfileOperation = createUserProfileOperation;
+    }
+
+    public void createOrUpdate(final UserProfile userProfile) throws BadRequestException {
+        final JurisdictionEntity existingJurisdiction = jurisdictionRepository.findEntityById(
```

Comment:
If you are happy with SaveUserProfileOperation then this method could be called saveUserProfile